File: Ordenação/Python/selectionSort.py
def selection(vt): # Função do SelectionSort, de parametro recebe um vetor.

	for i in range(0,len(vt)): # Para i no alcance de 0 até o tamanho do vetor.

		posMen = i # Define a posição do 'menor valor' sendo o valor de "i" posição inicial.
		valMen = vt[i] # Define o valor do 'menor valor' sendo o valor de "i" posição inicial.

		for j in range(i+1,len(vt)): # Para j no alcance de i+1 (n precisa analisar ele mesmo, caso j=0) até o tamanho do vetor.
			
			if(valMen > vt[j]): # Se o valor da posição "i" for maior que o valor do vet na pos "j".
				posMen = j # Troca "posMen" para a posição do novo 'menor valor'.
				valMen = vt[j] # Troca "valMen" para o valor do novo 'menor valor'.
			
		# Faz a troca dos valores pela seleção.
		aux = vt[i]
		vt[i] = vet[posMen]
		vet[posMen] = aux

vet = [] # Inicializa a Lista/Vetor.
n = int(input("Digite o Tamanho do Vetor: ")) # "Restringe" o Tamanho de elementos.

for t in range(0,n): # Para t no alcance de 0 até o número inserido pelo user em "n".
	val = int(input("Digite o {}º elemento: ".format(t+1))) # Pega do usuario o valor pra add no vetor
	vet.append(val) # Adiciona o valor digitado no vetor

print("Vetor antes do SelectionSort: {}".format(vet))

selection(vet) # Faz o SelectionSort

print("Vetor depois do SelectionSort: {}".format(vet))

File: Ordenação/Cpp/selectionSort.cpp
#include <iostream>

using namespace std;

int main(){
	
	int posMen, valMen, aux, aux2, n;

	cout << "Digite o tamanho do vetor: ";
	cin >> n;

	int vet[n];

	for(int k=0; k<n; k++){
		cout << "Digite o " << k+1 << "º" << " elemento: ";
		cin >> vet[k];
	}


	cout << "Array antes do SelectionSort: " << endl;

	for(int a=0; a<n; a++){ 
		cout << vet[a] << " ";
	}

	cout << endl;

	// --- Crescente ---


	for(int i=0; i<n; i++){

		posMen = i;
		valMen = vet[i];

		for(int j=i+1; j<n; j++){

			if(valMen > vet[j]){
				posMen = j;
				valMen = vet[j];
			}


		}

		aux = vet[i];
		vet[i] = vet[posMen];
		vet[posMen] = aux;

	}

	cout << "Array antes do SelectionSort Crescente: " << endl;

	for(int dc=0; dc<n; dc++){ 
		cout << vet[dc] << " ";
	}

	cout << endl;


	// --- Decrescente ---

	for(int l=0; l<n; l++){

		posMen = l;
		valMen = vet[l];

		for(int c=l+1; c<n; c++){

			if(valMen < vet[c]){
				posMen = c;
				valMen = vet[c];
			}


		}

		aux2 = vet[l];
		vet[l] = vet[posMen];
		vet[posMen] = aux2;

	}

	cout << "Array depois do SelectionSort Decrescente: " << endl;

	for(int dd=0; dd<n; dd++){ 
		cout << vet[dd] << " ";
	}

	cout << endl;

	return 0;
}